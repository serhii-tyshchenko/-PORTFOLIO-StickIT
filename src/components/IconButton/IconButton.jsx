import { iconButtonPropTypes, iconButtonDefaultProps } from './IconButton.props';

function IconButton({ icon, onClick, extraClassName, title, children }) {
  const componentClassName = `icon-${icon} ${extraClassName}`;

  return (
    <button className={componentClassName} onClick={onClick} title={title} type="button">
      {children}
    </button>
  );
}

IconButton.propTypes = iconButtonPropTypes;
IconButton.defaultProps = iconButtonDefaultProps;

export default IconButton;
