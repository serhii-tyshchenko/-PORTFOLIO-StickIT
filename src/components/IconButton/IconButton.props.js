import PropTypes from 'prop-types';

export const iconButtonPropTypes = {
  icon: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  extraClassName: PropTypes.string,
  title: PropTypes.string,
  children: PropTypes.node,
};

export const iconButtonDefaultProps = {
  extraClassName: '',
  title: '',
  children: null,
};
