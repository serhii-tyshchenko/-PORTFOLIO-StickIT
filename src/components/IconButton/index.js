/* eslint-disable import/prefer-default-export */
export { default as IconButton } from './IconButton';
