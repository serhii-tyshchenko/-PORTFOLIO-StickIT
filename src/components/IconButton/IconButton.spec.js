import renderer from 'react-test-renderer';

import IconButton from './index';

describe('IconButton', () => {
  it('renders correctly', () => {
    const props = {
      onClick: jest.fn(),
      icon: 'plus',
    };
    const component = renderer.create(<IconButton {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
