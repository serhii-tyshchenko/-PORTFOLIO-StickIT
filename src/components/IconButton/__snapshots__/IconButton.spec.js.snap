// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders correctly 1`] = `
<button
  className="icon-plus"
  onClick={[MockFunction]}
  title=""
  type="button"
/>
`;
